File: src/net/reichholf/dreamdroid/helpers/enigma2/Movie.java
/* © 2010 Stephan Reichholf <stephan at reichholf dot net>
 * 
 * Licensed under the Create-Commons Attribution-Noncommercial-Share Alike 3.0 Unported
 * http://creativecommons.org/licenses/by-nc-sa/3.0/
 */

package net.reichholf.dreamdroid.helpers.enigma2;

import java.util.ArrayList;

import org.apache.http.NameValuePair;
import org.apache.http.message.BasicNameValuePair;

import net.reichholf.dreamdroid.helpers.ExtendedHashMap;

/**
 * @author sreichholf
 * 
 */
public class Movie{
	public static final String REFERENCE = "reference";
	public static final String TITLE = "title";
	public static final String DESCRIPTION = "description";
	public static final String DESCRIPTION_EXTENDED = "descriptionEx";
	public static final String SERVICE_NAME = "servicename";
	public static final String TIME = "time";
	public static final String TIME_READABLE = "time_readable";
	public static final String LENGTH = "length";
	public static final String TAGS = "tags";
	public static final String FILE_NAME = "filename";
	public static final String FILE_SIZE = "filesize";
	public static final String FILE_SIZE_READABLE = "filesize_readable";

	public static ArrayList<NameValuePair> getDeleteParams(ExtendedHashMap movie){
		ArrayList<NameValuePair> params = new ArrayList<NameValuePair>();
		params.add(new BasicNameValuePair("sRef", movie.getString(Movie.REFERENCE)));
		
		return params;
	}
	
}

File: app/src/net/reichholf/dreamdroid/helpers/enigma2/Movie.kt
/* © 2010 Stephan Reichholf <stephan at reichholf dot net>
 *
 * Licensed under the Create-Commons Attribution-Noncommercial-Share Alike 3.0 Unported
 * http://creativecommons.org/licenses/by-nc-sa/3.0/
 */

package net.reichholf.dreamdroid.helpers.enigma2

import net.reichholf.dreamdroid.enigma.Movie as TypedMovie
import net.reichholf.dreamdroid.helpers.NameValuePair

/**
 * XML field names and request helpers for movies. UI uses
 * [net.reichholf.dreamdroid.enigma.Movie].
 */
object Movie {
    const val KEY_REFERENCE: String = Service.KEY_REFERENCE
    const val KEY_TITLE: String = "title"
    const val KEY_DESCRIPTION: String = "description"
    const val KEY_DESCRIPTION_EXTENDED: String = "descriptionEx"
    const val KEY_SERVICE_NAME: String = Service.KEY_NAME
    const val KEY_TIME: String = "time"
    const val KEY_TIME_READABLE: String = "time_readable"
    const val KEY_LENGTH: String = "length"
    const val KEY_TAGS: String = "tags"
    const val KEY_FILE_NAME: String = "filename"
    const val KEY_FILE_SIZE: String = "filesize"
    const val KEY_FILE_SIZE_READABLE: String = "filesize_readable"

    fun getDeleteParams(movie: TypedMovie): ArrayList<NameValuePair> {
        val params = ArrayList<NameValuePair>()
        params.add(NameValuePair("sRef", movie.reference))
        return params
    }
}
